kratos/utilities/result_dabatase.h

```diff
@@ -0,0 +1,706 @@
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    DataType mData; // The database storing the values
+
+    ///@}
+}; // Class VariableDatabase
+
+/**
+ * @class ResultDatabase
+ * @ingroup KratosCore
+ * @brief This class stores the results of a simulation for a later comparison
+ * @details The results are stored in a map which stores the results in tables. The following structure is followed:
+ *  - The key is the Id of the variable considered (so it does not depend if a double or component)
+ *  - A vector containing a set of tables. This tables are which actually contain the results
+ * @note If the table could store N columns without requiring a template argument the vector would not be required, saving NvariablesxNColumnsxNsteps doubles of memory
+ * @author Vicente Mataix Ferrandiz
+*/
+class ResultDatabase
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ResultDatabase
+    KRATOS_CLASS_POINTER_DEFINITION(ResultDatabase);
+
+    /// The definition of the index type
+    typedef std::size_t IndexType;
+
+    /// The definition of the sizetype
+    typedef std::size_t SizeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor.
+     */
+    ResultDatabase(){}
+
+    /// Destructor.
+    virtual ~ResultDatabase()
+    {
+        this->Clear();
+    }
+
+    /// Copy constructor.
+    ResultDatabase(ResultDatabase const& rOther)
+        : mData(rOther.mData),
+          mCommonColumn(rOther.mCommonColumn)
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method initializes the database
+     * @param rVariablesIndexes The list of indexes of the variables
+     * @param NumberOfEntites The number of entities considered
+     */
+    void Initialize(
+        const std::vector<IndexType>& rVariablesIndexes,
+        const std::vector<IndexType>& rValuesSizes,
+        const SizeType NumberOfEntites,
+        const SizeType NumberOfGP = 1
+        )
+    {
+        // If no variables we skip
+        if (rVariablesIndexes.size() == 0) {
+            return void();
+        }
+
+        KRATOS_ERROR_IF_NOT(rVariablesIndexes.size() == rValuesSizes.size()) << "Inconsistent sizes in the values sizes and the variable indexes" << std::endl;
+
+        // Auxiliar lambda to generate vectors of tables
+        auto table_generator =[](const SizeType NumberOfEntites, const SizeType NumberOfComponents, const SizeType NumberOfGP){std::vector<Table<double, double>*> aux_1(NumberOfComponents, nullptr); EntityDatabase aux_2(NumberOfGP, aux_1); VariableDatabase data(NumberOfEntites, aux_2); for (IndexType k = 0; k < NumberOfEntites; ++k){for (IndexType j = 0; j < NumberOfGP; ++j){ for (IndexType i = 0; i < NumberOfComponents; ++i){ data[k][j][i] = new Table<double, double>();}}}; return data;};
```

Comment:
this is unreadable, please refactor

btw you are allocating with `new`, but I cannot see the corresponding `delete`